starknet_py/net/account/account.py

```diff
@@ -359,6 +360,60 @@ async def verify_message(self, typed_data: TypedData, signature: List[int]) -> b
         message_hash = typed_data_dataclass.message_hash(account_address=self.address)
         return await self._verify_message_hash(message_hash, signature)
 
+    @staticmethod
+    async def deploy_account(
+        *,
+        address: int,
+        class_hash: int,
+        salt: int,
+        key_pair: KeyPair,
+        client: Client,
+        chain: StarknetChainId,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> "Account":
+        """
+        Deploys an account contract with provided class_hash on StarkNet and returns
+        an Account instance to be used.
+
+        Provided address must be first prefunded with enough tokens, otherwise the method will fail.
+
+        :param address: calculated and prefunded address of the new account.
+        :param class_hash: class_hash of the account contract to be deployed.
+        :param salt: salt used to calculate the address.
+        :param client: a Client instance used for deployment.
+        :param key_pair: KeyPair used to calculate address and sign deploy account transaction.
+        :param chain: id of the StarkNet chain used.
+        :param max_fee: max fee to be paid for deployment, must be less or equal to the amount of tokens prefunded.
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
+        """
+        if address != (
+            computed := compute_address(
+                salt=salt,
+                class_hash=class_hash,
+                constructor_calldata=[key_pair.public_key],
+                deployer_address=0,
+            )
+        ):
+            raise ValueError(
+                f"Provided address {hex(address)} is different than computed address {hex(computed)} "
+                f"for the given class_hash and salt"
+            )
+
+        account = Account(
+            address=address, client=client, key_pair=key_pair, chain=chain
+        )
+        deploy_account_tx = await account.sign_deploy_account_transaction(
+            class_hash=class_hash,
+            contract_address_salt=salt,
+            constructor_calldata=[key_pair.public_key],
```

Comment:
We do not support all accounts here. Maybe we should allow passing constructor_calldata as an argument and default to this one?

Reply:
That's true - we don't have any class hash hardcoded here. In reality you want to deploy account behind the proxy. At minimum it will take both the public key and class hash.

Reply:
Okay, we can accept any calldata.